src/test/net_tests.cpp

```diff
@@ -72,6 +73,18 @@ CDataStream AddrmanToStream(CAddrManSerializationMock& _addrman)
 
 BOOST_FIXTURE_TEST_SUITE(net_tests, BasicTestingSetup)
 
+BOOST_AUTO_TEST_CASE(cnode_listen_port)
+{
+    // test default
+    unsigned short port = GetListenPort();
+    BOOST_CHECK(port == Params().GetDefaultPort());
+    // test set port
+    unsigned short altPort = 12345;
+    SoftSetArg("-port", std::to_string(altPort));
```

Comment:
This will modify the global state of test run after this test case. There are other places where this happnes, but imo, unit tests should really be unit test.